File: src/prog_langs/python/foreach_search.py
searched = []# 空の配列を作成
array    = ['test', 'hoge', 'piyo', 'huga']

for val in (f for f in array if 'e' in f):
    # eが含まれる場合のみ空の配列にappend
    searched.append(val)

for v in searched:# 全て回す
    print(v)

# 実行結果
# test
# hoge

File: src/prog_langs/php/foreach_search.php
<?php
	$x = ['test','hoge','piyo','huga'];// 元となる配列
	$search = [];// 検索データを格納する空の配列

	foreach ($x as $key => $value) {
		if(strpos($value,'e') !== false){// データにeが含まれる場合
			$search[] = $value;// $searchに格納する
		}
	}

	foreach ($search as $key => $value) {// 作成した$search配列を回す
		echo $value."\n";// すべてecho
	}
	// 実行結果
	// test
	// hoge
?>